core/src/main/java/com/linecorp/armeria/common/stream/StreamWriter.java

```diff
@@ -74,7 +98,14 @@ default void write(Supplier<? extends T> o) {
 
     /**
      * Writes the specified object {@link Supplier} to the {@link StreamMessage}. The object provided by the
-     * {@link Supplier} will be transferred to the {@link Subscriber}.
+     * {@link Supplier} will be transferred to the {@link Subscriber}. Note: If the object provided by the
+     * {@link Supplier} is a {@link ReferenceCounted} such as {@link ByteBuf} and {@link ByteBufHttpData},
+     * it will be released on any of the following events:
```

Comment:
Is there any event that doesn't release it? I think it might be simpler to just say ownership is transferred since there doesn't seem to be a situation it'd be safe for the user not to retain if they want to keep using the object.

Reply:
Ownership is not transferred completely if a user retains it, so I wasn't sure if it's enough to just say so. Let me try to change this a little bit so that it's clear that the object will always be released.